### PR TITLE
[ML] Start and stop model deployments

### PR DESCRIPTION
**Pull request by dimitris-athanasiou**

### CODE REVIEW

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportStopTrainedModelDeploymentAction.java [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+        logger.debug("[{}] Received request to undeploy", request.getId());
+
+        ActionListener<GetTrainedModelsAction.Response> getModelListener = ActionListener.wrap(
+            getModelsResponse -> {
+                List<TrainedModelConfig> models = getModelsResponse.getResources().results();
+                if (models.isEmpty()) {
+                    listener.onResponse(new StopTrainedModelDeploymentAction.Response(true));
+                    return;
+                }
+                if (models.size() > 1) {
```
Comment (davidkyle) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportStopTrainedModelDeploymentAction.java:85]: In future we may have more than one model config using the deployment. I might be that we don't do the `GetTrainedModelsAction` here and just look for persistent tasks that match the model ID

**Review by davidkyle — APPROVED** [review]
LGTM

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/StartTrainedModelDeploymentAction.java [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+        @Override
+        public String toString() {
+            return Strings.toString(this);
+        }
+    }
+
+    public static class TaskParams implements PersistentTaskParams {
+
+        public static final Version VERSION_INTRODUCED = Version.V_7_13_0;
```
[the remark this answers is not in the document]
Comment (dimitris-athanasiou) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/StartTrainedModelDeploymentAction.java:120]: Will change to 8 and we see :-)

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportStopTrainedModelDeploymentAction.java (hunk 1 above)
Comment (dimitris-athanasiou) [reply x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportStopTrainedModelDeploymentAction.java:85]: Agreed. That's how we typically implement stop actions where we handle stopping multiple tasks at once. Just thought this was simpler for now.

**Pull request merged**